### PR TITLE
Remove conditional code supporting old Qt5 versions

### PR DESCRIPTION
**Pull request by winterz**
Remove support for Qt5 versions less than 5.15.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ CMake option `CMAKE_PREFIX_PATH` and point it to the folder of your installation
 For example:
 
 ```bash
-    cmake -DCMAKE_PREFIX_PATH=$HOME/Qt/5.11.2/gcc_64 ..
+    cmake -DCMAKE_PREFIX_PATH=$HOME/Qt/5.15.4/gcc_64 ..
 ```
```
Comment (ferdnyc) [inline INSTALL.md:50]: Maybe this should just use `$HOME/Qt/6.6.0/gcc_64` as an example, instead? (Along with the envvar version below.)

Doesn't hurt to _encourage_ users to migrate to Qt6, after all.

**Review by ferdnyc — COMMENTED** [review]
Wow, I had _no **idea**_ there was so MUCH legacy code in there. Hooray for streamlining! :+1:

File: common/recursiveproxymodelbase.h [hunk 2]
```diff
@@ -22,28 +22,19 @@
...
 
-#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
-#include <kde/krecursivefilterproxymodel.h>
-#define GAMMARAY_PROXY_BASE_CLASS KRecursiveFilterProxyModel
-#else
 #include <QSortFilterProxyModel>
 #define GAMMARAY_PROXY_BASE_CLASS QSortFilterProxyModel
-#endif
 
 class GAMMARAY_COMMON_EXPORT RecursiveProxyModelBase : public GAMMARAY_PROXY_BASE_CLASS
```
Comment (Waqar144) [inline common/recursiveproxymodelbase.h:28]: `GAMMARAY_PROXY_BASE_CLASS` -> `QSortFilterProxyModel`

File: common/recursiveproxymodelbase.h [hunk 3]
```diff
@@ -22,28 +22,19 @@
...
-#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
-    using KRecursiveFilterProxyModel::KRecursiveFilterProxyModel;
-#else
     RecursiveProxyModelBase(QObject *parent)
         : QSortFilterProxyModel(parent)
     {
         setRecursiveFilteringEnabled(true);
     }
 
     virtual bool acceptRow(int sourceRow, const QModelIndex &sourceParent) const;
```
Comment (Waqar144) [inline common/recursiveproxymodelbase.h:37]: This function can be removed as well in favor of the one below it, but we can do this as a followup

**Review by Waqar144 — CHANGES_REQUESTED** [review]

File: INSTALL.md (hunk 1 above)
Comment (winterz) [reply INSTALL.md:50]: good idea. I'll do that

File: common/recursiveproxymodelbase.h (hunk 2 above)
Comment (winterz) [reply common/recursiveproxymodelbase.h:28]: will do

Comment (winterz) [reply common/recursiveproxymodelbase.h]: done

File: common/recursiveproxymodelbase.h (hunk 3 above)
Comment (winterz) [reply common/recursiveproxymodelbase.h:36]: agreed

**Review by Waqar144 — APPROVED** [review]

**Pull request merged**